app/services/step_content_parser.rb

```diff
@@ -87,16 +87,23 @@ def bulleted_list?(section)
 
   def link_content(section)
     section.map do |line|
-      if line.scan(/\[/).empty?
-        { "text": line[2..-1] }
-      elsif /\[(?<text>(.+))\]\((?<href>(.+))\)((?<context>.*))$/ =~ line
-        payload = {
-          "text": text,
-          "href": href
-        }
-        payload[:context] = context.strip unless context.blank?
-        payload
+      payload = {
```

Comment:
There's a couple of nested `if` statements, with the `else` outside of the `if`, which made it difficult to fully understand what the different scenarios are.

Could this be written as (for example):
```
  def link_content(section)
    section.map do |line|
      payload = {}
      if /\[(?<text>(.+))\]\((?<href>(.+))\)((?<context>.*))$/ =~ line && safely_parse_path(href)
        payload = {
          "text": text,
          "href": href,
        }
        payload[:context] = context.strip unless context.blank?
      else
        payload[:text] = line[2..-1]
      end
      payload
    end
  end
```

If the assumption I've made that the two conditionals can be joined with an `and` is incorrect, then could a test be added to cater for cases where the first condition is true but the second is false.  My assumption has resulted in all the tests passing with the proposed code.

Reply:
Good idea, will rewrite 👍 wasn't sure about using the `href` variable in the same conditional as it is created in.